src/main/kotlin/org/openrewrite/kotlin/KotlinTypeMapping.kt

```diff
@@ -882,6 +886,8 @@ class KotlinTypeMapping(typeCache: JavaTypeCache, firSession: FirSession) : Java
             // TODO: fix type attribution for anonymous objects. Currently, returns JavaType.Unknown.
             if (symbol.dispatchReceiverType != null) {
                 resolvedOwner = type(symbol.dispatchReceiverType)
+            } else if (ownerFallBack is FirFunctionSymbol<*>) {
+                resolvedOwner = methodDeclarationType(ownerFallBack.fir, null, firFileSymbol)
```

Comment:
The signature is set correctly to the method declaration signature. This change fixes the JavaType returned by the signature.